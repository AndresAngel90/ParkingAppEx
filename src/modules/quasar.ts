import { Quasar, Dialog, Notify } from 'quasar'
import type { App } from 'vue'

import 'quasar/src/css/index.sass'
import '@quasar/extras/material-icons/material-icons.css'


export const install = (app: App) => {
  app.use(Quasar, {
    plugins: { Dialog, Notify }
  })
};
