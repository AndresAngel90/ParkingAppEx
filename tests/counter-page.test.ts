
import { mount } from '@vue/test-utils'
import { test, expect } from 'vitest'
import { Quasar } from 'quasar'

import CounterPage from '~/pages/index.vue'

const wrapperFactory = () => mount(CounterPage, {
  global: {
    plugins: [Quasar]
  }
})

test('mount component', () => {
  expect(CounterPage).toBeTruthy()

  const wrapper = wrapperFactory()
  const count = wrapper.get('[data-id="counter"]')

  expect(count.text()).toContain('0')
  expect(wrapper.html()).toMatchSnapshot()
})

test('handle + click', async () => {
  const wrapper = wrapperFactory();
  const count = wrapper.get('[data-id="counter"]');
  const addBtn = wrapper.get('[data-id="add"]');

  await addBtn.trigger('click')

  expect(count.text()).toContain('1')
  
  await addBtn.trigger('click')
  
  expect(count.text()).toContain('2')
  expect(wrapper.html()).toMatchSnapshot()
})

