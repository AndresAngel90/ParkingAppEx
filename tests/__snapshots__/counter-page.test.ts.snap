// Vitest Snapshot v1

exports[`handle + click 1`] = `
"<div class=\\"row\\">
  <div class=\\"column col-3 q-mx-auto q-mt-lg\\"><span class=\\"text-center text-subtitle2 q-my-sm\\"> Count: <span data-id=\\"counter\\">2</span></span><button class=\\"q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle q-btn--actionable q-focusable q-hoverable\\" tabindex=\\"0\\" type=\\"button\\" data-id=\\"add\\"><span class=\\"q-focus-helper\\"></span><span class=\\"q-btn__content text-center col items-center q-anchor--skip justify-center row\\">+</span><span class=\\"q-ripple\\" dir=\\"ltr\\"><span class=\\"q-ripple__inner\\" style=\\"height: 0px; width: 0px; transform: translate3d(0px,0px,0) scale3d(.2,.2,1);\\"></span></span><span class=\\"q-ripple\\" dir=\\"ltr\\"><span class=\\"q-ripple__inner\\" style=\\"height: 0px; width: 0px; transform: translate3d(0px,0px,0) scale3d(.2,.2,1);\\"></span></span></button></div>
</div>"
`;

exports[`mount component 1`] = `
"<div class=\\"row\\">
  <div class=\\"column col-3 q-mx-auto q-mt-lg\\"><span class=\\"text-center text-subtitle2 q-my-sm\\"> Count: <span data-id=\\"counter\\">0</span></span><button class=\\"q-btn q-btn-item non-selectable no-outline q-btn--standard q-btn--rectangle q-btn--actionable q-focusable q-hoverable\\" tabindex=\\"0\\" type=\\"button\\" data-id=\\"add\\"><span class=\\"q-focus-helper\\"></span><span class=\\"q-btn__content text-center col items-center q-anchor--skip justify-center row\\">+</span></button></div>
</div>"
`;
